include/logging/log.h

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LOG_H_
+#define LOG_H_
+
+#include <logging/log_instance.h>
+#include <logging/log_core.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define LOG_LEVEL_NONE  0
+#define LOG_LEVEL_ERR   1
+#define LOG_LEVEL_WARN  2
+#define LOG_LEVEL_INFO  3
+#define LOG_LEVEL_DBG   4
+
+/**
+ * @brief Writes an ERROR level message to the log.
+ *
+ * @details It's meant to report severe errors, such as those from which it's
+ * not possible to recover.
+ *
+ * @param ... A string optionally containing printk valid conversion specifier,
+ * followed by as many values as specifiers.
+ */
+#define LOG_ERR(...)    LOG_INTERNAL_MODULE(LOG_LEVEL_ERR, __VA_ARGS__)
+
+
+/**
+ * @brief Writes an WARNING level message to the log.
+ *
+ * @details It's meant to register messages related to unusual situations that
+ * are not necessarily errors.
+ *
+ * @param ... A string optionally containing printk valid conversion specifier,
+ * followed by as many values as specifiers.
+ */
+#define LOG_WARN(...)   LOG_INTERNAL_MODULE(LOG_LEVEL_WARN, __VA_ARGS__)
+
+/**
+ * @brief Writes an INFO level message to the log.
+ *
+ * @details It's meant to write generic user oriented messages.
+ *
+ * @param ... A string optionally containing printk valid conversion specifier,
+ * followed by as many values as specifiers.
+ */
+#define LOG_INFO(...)   LOG_INTERNAL_MODULE(LOG_LEVEL_INFO, __VA_ARGS__)
+
+/**
+ * @brief Writes an DEBUG level message to the log.
+ *
+ * @details It's meant to write developer oriented information.
+ *
+ * @param ... A string optionally containing printk valid conversion specifier,
+ * followed by as many values as specifiers.
+ */
+#define LOG_DBG(...)    LOG_INTERNAL_MODULE(LOG_LEVEL_DBG, __VA_ARGS__)
+
+/**
+ * @brief Writes an ERROR level message to the log.
+ *
+ * It's meant to report severe errors, such as those from which it's
+ * not possible to recover.
+ *
+ * @param inst Pointer to the log filter associated with the instance.
+ * @param ... A string optionally containing printk valid conversion specifier,
+ * followed by as many values as specifiers.
+ */
+#define LOG_INST_ERR(inst, ...)	\
+	LOG_INTERNAL_INSTANCE(LOG_LEVEL_ERR, _inst, __VA_ARGS__)
```

Comment:
Is `_inst` (vs `inst`) intended in the expansion of the macro?

Reply:
fixed.